databricks/koalas/indexing.py

```diff
@@ -400,16 +395,56 @@ def _select_rows(self, rows_sel):
                 # If slice is None - select everything, so nothing to do
                 return None, None
             elif len(self._internal.index_columns) == 1:
+                sdf = self._kdf_or_kser._internal.sdf
+                index = self._kdf_or_kser.index
+                index_column = index.to_series()
+                index_data_type = index_column.spark_type
                 start = rows_sel.start
                 stop = rows_sel.stop
```

Comment:
Can we remove this and use `rows_sel.start` and `rows_sel.stop`? Seems confusing what `start` and `stop` have inside because it's later changed to the position in the natural order.